crates/vm/levm/src/db/gen_db.rs

```diff
@@ -231,8 +223,61 @@ impl<'a> VM<'a> {
         key: H256,
         new_value: U256,
     ) -> Result<(), VMError> {
+        self.backup_storage_slot(address, key)?;
+
         let account = self.get_account_mut(address)?;
         account.storage.insert(key, new_value);
         Ok(())
     }
+
+    pub fn backup_storage_slot(&mut self, address: Address, key: H256) -> Result<(), VMError> {
+        let value = self.get_storage_value(address, key)?;
+
+        let account_storage_backup = self
+            .current_call_frame_mut()?
+            .call_frame_backup
+            .original_account_storage_slots
+            .entry(address)
+            .or_insert(HashMap::new());
+
+        if account_storage_backup.get(&key).is_none() {
+            account_storage_backup.insert(key, value);
+        }
+
+        Ok(())
+    }
+
+    pub fn backup_account_info(&mut self, address: Address) -> Result<(), VMError> {
+        if self.call_frames.is_empty() {
+            return Ok(());
+        }
+
+        let maybe_account_info_backup = self
+            .current_call_frame_mut()?
+            .call_frame_backup
+            .original_accounts_info
+            .get_mut(&address);
+
+        if maybe_account_info_backup.is_none() {
```

Comment:
```suggestion
        let is_not_backed_up = self
            .current_call_frame_mut()?
            .call_frame_backup
            .original_accounts_info
            .get(&address)
            .is_none();

        if is_not_backed_up {
```
I would reframe it to something like that. The `get_mut` is a little bit confusing.

Reply:
I like it, tried your suggestion but clippy suggested replacing the `get -> is_none` with a `contains_key` and a negation so went with that here [7a477db](https://github.com/lambdaclass/ethrex/pull/2699/commits/7a477db319b17454bd04dd598d5dacf4eff84263)

Reply:
Nice! Even better